src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -617,98 +626,141 @@ func (s *peersSource) readIndex(
 	// FOLLOWUP(r): Try to reuse any metadata fetched during the ReadData(...)
 	// call rather than going to the network again
 	r := result.NewIndexBootstrapResult()
+	builders := make(result.IndexBuilders)
 	if shardsTimeRanges.IsEmpty() {
 		return r, nil
 	}
 
-	session, err := s.opts.AdminClient().DefaultAdminSession()
-	if err != nil {
-		s.log.Error("peers bootstrapper cannot get default admin session", zap.Error(err))
-		r.SetUnfulfilled(shardsTimeRanges)
-		return nil, err
-	}
-
 	var (
-		count         = len(shardsTimeRanges)
-		concurrency   = s.opts.DefaultShardConcurrency()
-		dataBlockSize = ns.Options().RetentionOptions().BlockSize()
-		resultOpts    = s.opts.ResultOptions()
-		idxOpts       = ns.Options().IndexOptions()
-		resultLock    = &sync.Mutex{}
-		wg            sync.WaitGroup
+		count          = len(shardsTimeRanges)
+		concurrency    = s.opts.DefaultShardConcurrency()
+		indexBlockSize = ns.Options().RetentionOptions().BlockSize()
```

Comment:
This might be `dataBlockSize`?  I think index block size would come from `ns.Options().IndexOptions().BlockSize()`?